app/controllers/triannon/annotations_controller.rb

```diff
@@ -106,7 +111,9 @@ def create
           format.xml {
             accept_return_type = mime_type_from_accept(["application/xml", "text/xml", "application/x-xml"])
             render :body => @annotation.graph.to_rdfxml, status: 201, content_type: accept_return_type if accept_return_type }
-          format.html { redirect_to @annotation }
+          format.html {
+            require 'cgi'
+            redirect_to "/annotations/#{params[:anno_root]}/#{CGI.escape(@annotation.id)}" }
```

Comment:
Can you use a named route here, and avoid needing to CGI escape it yourself, e.g. `redirect_to annotation_path(anno_root: 'abc', id: 'def')`

If not, I think `ERB::Util.url_escape` [1] may do the same thing, and I think Rails will have provided it for you already.

[1] http://ruby-doc.org/stdlib-1.9.3/libdoc/erb/rdoc/ERB/Util.html